pkg/oc/lib/newapp/cmd/newapp.go

```diff
@@ -1180,6 +1185,35 @@ func (c *AppConfig) followRefToDockerImage(ref *kapi.ObjectReference, isContext
 	return c.followRefToDockerImage(target, isContext, objects)
 }
 
+func (c *AppConfig) checkImageStreamsAndTags(objects app.Objects) app.Objects {
+	var tagsToRemove []int
+	for i, obj := range objects {
+		if ist, ok := obj.(*imageapi.ImageStreamTag); ok {
+			istNamespace := ist.Namespace
+			if len(istNamespace) == 0 {
+				istNamespace = c.OriginNamespace
+			}
+			streamName, tagName, ok := imageapi.SplitImageStreamTag(ist.Name)
+			if ok {
+				is := c.findImageStreamInObjectList(objects, streamName, istNamespace)
+				if is != nil {
+					for name := range is.Spec.Tags {
+						if name == tagName {
+							tagsToRemove = append(tagsToRemove, i)
+						}
+					}
+				}
+			}
+
+		}
+	}
+	sort.Sort(sort.Reverse(sort.IntSlice(tagsToRemove)))
+	for _, objIndex := range tagsToRemove {
+		objects = append(objects[:objIndex], objects[objIndex+1:]...)
+	}
```

Comment:
this looks overly cute (and is probably incorrect.  each time you change objects, objIndex becomes invalid for the next reference).  Just iterate the original list and copy the ones that aren't in "tagsToRemove" into a new list.

Reply:
updated in current push